lib/go/templates/marketV2_templates.go

```diff
@@ -0,0 +1,159 @@
+package templates
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go-sdk"
+)
+
+// This contains template transactions for the second version of the Top Shot
+// marketplace, which uses a capability to access the owner's moment collection
+// instead of storing the moments in the sale collection directly
+
+// GenerateCreateSaleV2Script creates a cadence transaction that creates a Sale collection
+// and stores in in the callers account published
+func GenerateCreateSaleV2Script(ftAddr, topshotAddr, marketAddr, beneficiaryAddr flow.Address, tokenStorageName string, cutPercentage float64) []byte {
+
+	template := `
+		import FungibleToken from 0x%[5]s
+		import TopShot from 0x%[6]s
+		import Market from 0x%[1]s
+
+		transaction {
+			prepare(acct: AuthAccount) {
+				let ownerCapability = acct.getCapability<&{FungibleToken.Receiver}>(/public/%[3]sReceiver)!
+				let beneficiaryCapability = getAccount(0x%[2]s).getCapability<&{FungibleToken.Receiver}>(/public/%[3]sReceiver)!
+
+				let ownerCollection: Capability<&TopShot.Collection> = acct.link<&TopShot.Collection>(/private/MomentCollection, target: /storage/MomentCollection)!
+
+				let collection <- Market.createSaleCollection(ownerCollection: ownerCollection, ownerCapability: ownerCapability, beneficiaryCapability: beneficiaryCapability, cutPercentage: %[4]f)
+				
+				acct.save(<-collection, to: /storage/topshotSaleCollection)
```

Comment:
shall we append a version to the storage path if we want to have both old version and new version functional at the same time? It would help managing the migration a lot too. cc @EricLin2004